### PR TITLE
V1.3 tracking due fees, tagging

### PR DESCRIPTION
**Pull request by lucdx**
V1.3 Fees Update

ListFees now sorts the fees according to the due date, with due fees appearing on top. And furthest dates at the bottom.

New ListDueFees Command now only shows the people with fees due with respect to the current date in the system, when the admin is executing the command.

ListDueFees adds a "feesdue" tag to the person to allow for searching in the future by tags, or for plain view when using listall/viewall commands. Editing of fees back to default values will remove this tag. 

Calling ListDueFees again will also check for the tags again if i.e, used on another session with a different current date.

Created new AdaptedFees for storage so Fees is no longer part of AdaptedPerson

Documentation will follow in another PR. 
UG complete,
DG WIP.

### CODE REVIEW

**Comment by micklwl** [comment]
Use the Utils.isDateValid method

File: src/seedu/addressbook/commands/CommandResult.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
+package seedu.addressbook.commands;
```
Comment (micklwl) [inline src/seedu/addressbook/commands/CommandResult.java]: Why is this file here?

Comment (lucdx) [reply src/seedu/addressbook/commands/CommandResult.java]: huh? It's here in the original also right?

File: src/seedu/addressbook/commands/fees/AddFeesCommand.java [hunk 2]
```diff
@@ -47,6 +51,16 @@ public CommandResult execute() {
             final ReadOnlyPerson target = getTargetReadOnlyPerson();
```
Comment (Meg1456) [inline src/seedu/addressbook/commands/fees/AddFeesCommand.java]: One of the changes that has been implemented in a previous patch is that getTargetPerson returns an editable Person, while getTargetReadOnlyPerson returns uneditable one now. Since the commands are already distinguished, we need not use both functions anymore. Can directly do Person person = getTargetPerson(); if you want the editable version.

File: src/seedu/addressbook/commands/fees/EditFeesCommand.java [hunk 3]
```diff
@@ -8,13 +11,14 @@
...
 /**
  * Adds fees to a respective person
  */
-public class AddFeesCommand extends Command {
+public class EditFeesCommand extends Command {
 
-    public static final String COMMAND_WORD = "addfees";
+    public static final String COMMAND_WORD = "editfees";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n" + "Adds fees to an existing person "
```
Comment (Meg1456) [inline src/seedu/addressbook/commands/fees/EditFeesCommand.java]: I think you missed to update the message usage here. From add fees to edit fees.

File: src/seedu/addressbook/commands/fees/AddFeesCommand.java (hunk 2 above)
Comment (lucdx) [reply src/seedu/addressbook/commands/fees/AddFeesCommand.java]: AWESOME. Thanks I missed that

File: src/seedu/addressbook/commands/fees/EditFeesCommand.java (hunk 3 above)
Comment (lucdx) [reply src/seedu/addressbook/commands/fees/EditFeesCommand.java]: This too!!

File: src/seedu/addressbook/data/AddressBook.java [hunk 4]
```diff
@@ -141,8 +142,15 @@ public void clearAllExam() {
...
+    public List<ReadOnlyPerson> listFeesPerson() {
+        return allPersons.listFees();
+    }
+
+    /**
+     * Loops through the list to get overdue fees
+     */
+    public List<ReadOnlyPerson> dueFeesPerson(String date) {
+        return allPersons.dueFees(date);
     }
```
Comment (0WN463) [inline src/seedu/addressbook/data/AddressBook.java]: Methods should sound like verbs

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 5]
```diff
@@ -117,20 +121,80 @@ public void remove(ReadOnlyPerson toRemove) throws PersonNotFoundException {
...
+                    if ("feesdue".equals(t.tagName)) {
+                        temp.remove(t);
+                    }
+                }
+                p.setTags(temp);
+            }
+            copy.sort(new CustomComparator());
+        }
+        return Collections.unmodifiableList(copy);
+    }
```
Comment (0WN463) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: Wut am I even looking at...

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 6]
```diff
@@ -117,20 +121,80 @@ public void remove(ReadOnlyPerson toRemove) throws PersonNotFoundException {
...
+     */
+    public List<ReadOnlyPerson> listFees() {
+        if (internalList.isEmpty()) {
+            return Collections.unmodifiableList(internalList);
+        }
+        List<Person> copy = new ArrayList<>(internalList);
+        copy.sort(new CustomComparator());
+        return Collections.unmodifiableList(copy);
+    }
+
```
Comment (0WN463) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: Same as above

File: src/seedu/addressbook/storage/jaxb/AdaptedFees.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+public class AdaptedFees {
+    @XmlElement(required = true)
+    private String value;
+
+    @XmlElement(required = true)
+    private String duedate;
+
+    @XmlAttribute(required = true)
+    private boolean isPrivate;
+    /**
```
Comment (0WN463) [inline src/seedu/addressbook/storage/jaxb/AdaptedFees.java]: Hmm maybe it'll be a good idea to remove the isPrivate since it should always be private

File: test/java/seedu/addressbook/FormatterTest.java [hunk 8]
```diff
@@ -57,6 +57,70 @@ public void formatPersonsAllDetailsFormat() throws Exception {
...
+                        + " "
+                        + NEWLINE)
+                        + String.format(" 3. Person 3" + NEWLINE
+                        + " {private Fees: 0.00 / 00-00-0000} " + NEWLINE
+                        + " "
+                        + NEWLINE + " ")
+                        + NEWLINE);
+        assertFormatterBehaviour(PersonListFormat.FEES_DETAILS);
+    }
+
```
Comment (0WN463) [inline test/java/seedu/addressbook/FormatterTest.java]: Maybe String.Format(%s, NEWLINE) might make it more readable
